internal/errors/errors.go

```diff
@@ -1,5 +1,53 @@
 package errors
 
-import "fmt"
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+)
 
-var ErrBadRequest = fmt.Errorf("unrecoginized identifier")
+// Request Errors
+type RequestError struct {
+	Err    error
+	Msg    string
+	Method string
+	Uri    string
+	Logger *slog.Logger
+}
+
+func (e *RequestError) Error() string {
+	return fmt.Sprintf("Request Error: %s", e.Msg)
+}
+
+var ErrInvalidId = fmt.Errorf("invalid identifier")
+
+func BadRequest(w http.ResponseWriter, r *http.Request, err error, logger *slog.Logger) {
+	logger.Error(err.Error(), slog.String("method", r.Method), slog.String("uri", r.URL.RequestURI()))
+	http.Error(w, err.Error(), http.StatusBadRequest)
+}
+
+// Database Errors
+type DatabaseError struct {
+	Err    error
+	Msg    string
+	Method string
+	Id     string
+}
+
+func (e *DatabaseError) Error() string {
+	return fmt.Sprintf("Msg: %s. Err: %s", e.Msg, e.Err)
+}
+
+func (e *DatabaseError) NoRows(w http.ResponseWriter, r *http.Request, err error, logger *slog.Logger) {
+	logger.Error(err.Error(), slog.String("method", r.Method), slog.String("uri", r.URL.RequestURI()))
+	http.Error(w, e.Msg, http.StatusBadRequest)
+}
+
+func ServerError(w http.ResponseWriter, r *http.Request, err error, logger *slog.Logger) {
```

Comment:
Do you think it makes sense to give these functions a receiver of `(app *application)`, so that you can skip passing the logger as an arg and just do `app.logger`?

Reply:
As per ongoing discussion: what about pulling logging into its own module to separate it from app entirely?